### PR TITLE
Policies: Adds deprecation policy

### PR DESCRIPTION
**Pull request by bergquist**
Based on the internal document that have been circulated for a while. 

I'm not sure what the best place would be but I figure this belongs more under contributing than docs? But more than happy to change it.

### CODE REVIEW

File: contribute/deprecation-policy.md
Comment (bergquist) [inline contribute/deprecation-policy.md:32]: I'll write up an issue and link to it before merging.

File: contribute/deprecation-policy.md
Comment (DanCech) [inline contribute/deprecation-policy.md:7]:
```suggestion
If we cannot remove a feature without introducing a breaking chance we should create a deprecation plan as listed below:
```

File: contribute/deprecation-policy.md
Comment (DanCech) [inline contribute/deprecation-policy.md:5]:
```suggestion
Going forward we will do our best to **not** make breaking changes, in order to make upgrading Grafana as easy and reliable as possible. From now on we will avoid making unexpected breaking changes to the customer experience and our ability to operate our service.
```

File: contribute/deprecation-policy.md
Comment (DanCech) [inline contribute/deprecation-policy.md:10]:
```suggestion
- Determine usage levels of the feature.
```

**Review by DanCech — COMMENTED** [review]
I like where this is headed! Added a few suggestions

File: contribute/deprecation-policy.md
Comment (bergquist) [reply contribute/deprecation-policy.md:5]: kind of passive voice 🤔 
yours is better but I might take another stab at this one

File: contribute/deprecation-policy.md
Comment (DanCech) [inline contribute/deprecation-policy.md:7]: Typo fix and stronger commitment
```suggestion
If we cannot remove a feature without introducing a breaking change we will create a deprecation plan as listed below:
```

File: contribute/deprecation-policy.md
Comment (DanCech) [inline contribute/deprecation-policy.md:22]: Not sure disablement is a word
```suggestion
## Grace period between announcement and disabling feature by default
```

**Review by DanCech — APPROVED** [review]
LGTM, if I were to add anything it would be guidance about the time between disable-by-default and full removal.

File: contribute/deprecation-policy.md
Comment (torkelo) [inline contribute/deprecation-policy.md:5]: This has always from day 1 been a major goal and focus (to make upgrading as easy and as pain-free as possible). So not sure what is changing (other than adding a formal policy). This seems to conflate unexpected breaking changes and deprecating features. 

I would rephrase this a bit. 

```suggestion
Major releases are when we will be making breaking changes, like changing behavior or removing a feature. 

We always do our best **not** to introduce breaking changes in order to make upgrading Grafana as easy and reliable as possible. However, at times we have to introduce a breaking change by changing behavior or by removing a feature.  

But in order to remove a feature and introduce a breaking change we need to create a deprecation plan. 
```

File: contribute/deprecation-policy.md
Comment (torkelo) [inline contribute/deprecation-policy.md:7]: nitpick: These two sentences seem to say almost the thing and make it seem like a list will follow.

File: contribute/deprecation-policy.md
Comment (torkelo) [inline contribute/deprecation-policy.md:9]: This is the third "a deprecation plan that includes" in row :smile:   (see the two sentences above)

**Review by torkelo — APPROVED** [review]

File: contribute/deprecation-policy.md
Comment (bergquist) [reply contribute/deprecation-policy.md]: Solved :)

**Review by timlevett — APPROVED** [review]
LGTM, thanks for putting this together.

**Pull request merged**
